### PR TITLE
Update contribution guidelines per latest development changes

### PR DESCRIPTION
**Pull request by karimnaaji**
- Update collaboration guidelines per latest dev changes for node 14
- Add .nvmrc file for developers using several node versions
- Improve node error handling by using explicit `engine` in `package.json`. Example situation
```
> node --version
v12.22.1

> yarn run start-debug 
yarn run v1.22.10
error mapbox-gl@2.4.0-dev: The engine "node" is incompatible with this module. Expected version "14.16.0". Got "12.22.1"
error Commands cannot run with an incompatible environment.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

> nvm use
Found '/Users/karimnaaji/dev/mapbox-gl-js/.nvmrc' with version <14.16.0>
Now using node v14.16.0 (npm v6.14.11)
```

Fixes https://github.com/mapbox/mapbox-gl-js/issues/10778

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -25,15 +25,15 @@ git clone git@github.com:mapbox/mapbox-gl-js.git
...
 ```
 
 ### Linux
 
-Install [git](https://git-scm.com/), [node.js](https://nodejs.org/) (version ^10.15), [GNU Make](http://www.gnu.org/software/make/), and libglew-dev
+Install [git](https://git-scm.com/), [node.js](https://nodejs.org/) version 14, [GNU Make](http://www.gnu.org/software/make/), and libglew-dev
 ```bash
-sudo apt-get update &&
+sudo apt-get update
+curl -sL https://deb.nodesource.com/setup_14.x | sudo bash -
```
Comment (rreusser) [inline CONTRIBUTING.md:36]: Works for me, but is nodesource the place to acquire node@14? I thought they were a very specific enterprise flavor of node.

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
     "url": "git://github.com/mapbox/mapbox-gl-js.git"
   },
   "engines": {
-    "node": ">=6.4.0"
+    "node": "14.16.0"
```
Comment (rreusser) [inline package.json]: Doesn't need a `>=` or `~` token or anything, does it?

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "url": "git://github.com/mapbox/mapbox-gl-js.git"
   },
   "engines": {
-    "node": ">=6.4.0"
+    "node": "14.15.1"
```
Comment (rreusser) [inline package.json:14]: Doesn't need a `>=` or `~` token or anything, does it?

**Review by rreusser — APPROVED** [review]
Nothing blocking, just a couple nitpicky questions.

File: CONTRIBUTING.md (hunk 1 above)
Comment (karimnaaji) [reply CONTRIBUTING.md:36]: I'm not sure, I've setup a ubuntu machine and this is what worked for me. I can look at alternatives

Comment (rreusser) [reply CONTRIBUTING.md:36]: 100% fine with me not to bother. It's linux-specific, so if this is the way you do it there and it works, it's 💯 okay with me. (nodesource is great, I'm only just familiar with them as an enterprise node company)

**Pull request merged**